src/emulator/storage/rules/manager.ts

```diff
@@ -1,54 +1,87 @@
 import * as chokidar from "chokidar";
-import * as fs from "fs";
 import { EmulatorLogger } from "../../emulatorLogger";
 import { Emulators } from "../../types";
-import { FirebaseError } from "../../../error";
 import { SourceFile } from "./types";
 import { StorageRulesIssues, StorageRulesRuntime, StorageRulesetInstance } from "./runtime";
+import { RulesConfig } from "..";
 
 /**
- * Loads and maintains a {@link StorageRulesetInstance} for a given source file. Listens for
- * changes to the file and updates the ruleset accordingly.
+ * Keeps track of rules source file(s) and generated ruleset(s), either one for all storage
+ * resources or different rules for different resources.
+ *
+ * Example usage:
+ *
+ * ```
+ * const rulesManager = createStorageRulesManager(initialRules);
+ * rulesManager.start();
+ * rulesManager.updateSourceFile(newRules);
+ * rulesManager.stop();
+ * ```
  */
-export class StorageRulesManager {
-  private _sourceFile?: SourceFile;
+export interface StorageRulesManager {
+  /** Sets source file for each resource using the most recent rules. */
+  start(): Promise<StorageRulesIssues>;
+
+  /**
+   * Retrieves the generated ruleset for the resource. Returns undefined if the resource is invalid
+   * or if the ruleset has not been generated.
+   */
+  getRuleset(resource: string): StorageRulesetInstance | undefined;
+
+  /**
+   * Updates the source file and, correspondingly, the file watcher and ruleset for the resource.
+   * Returns an array of errors and/or warnings that arise from loading the ruleset.
+   */
+  updateSourceFile(rules: SourceFile, resource: string): Promise<StorageRulesIssues>;
```

Comment:
Discussed offline, we may want to remove this method as we don't need to support modifying currently running `StorageRulesManagers`.

Reply:
Yup, will do in follow up PR.